lib/components/widgets/loginScreen/login_screen.dart

```diff
@@ -1,17 +1,94 @@
+import 'dart:convert';
 import 'package:flutter/material.dart';
 import 'package:flutter/gestures.dart';
+import 'package:http/http.dart' as http;
 import 'package:classroom_app/components/styles/color/colors.dart';
-import 'package:classroom_app/components/widgets/CustomButton/custom_button.dart';
 import 'package:classroom_app/components/widgets/SignUpScreen/sign_up_screen.dart';
 
-class LoginScreen extends StatelessWidget {
+class LoginScreen extends StatefulWidget {
   const LoginScreen({super.key});
 
+  @override
+  State<LoginScreen> createState() => _LoginScreenState();
+}
+
+class _LoginScreenState extends State<LoginScreen> {
+  final _usernameController = TextEditingController();
+  final _passwordController = TextEditingController();
+  bool _isLoading = false;
+
+  Future<void> _login() async {
+    setState(() {
+      _isLoading = true;
+    });
+
+    try {
+      final response = await http.post(
+        Uri.parse('http://10.0.2.2:8080/login'),
+        headers: {
```

Comment:
Consider defining your API URLs as constants or in a configuration file for easier maintenance.